From a84d2cf8882c97574f1a787596bb357b199f4d85 Mon Sep 17 00:00:00 2001
From: Anil Kumar <anilk4.v@gmail.com>
Date: Mon, 4 Feb 2013 20:25:07 +0530
Subject: [PATCH 39/48] ARM: dts: omap3-devkit8000: Enable audio support

Add the needed sections to enable audio support on
Devkit8000 when booted with DT blob.

Signed-off-by: Anil Kumar <anilk4.v@gmail.com>
Signed-off-by: Benoit Cousson <benoit.cousson@linaro.org>
---
 arch/arm/boot/dts/omap3-devkit8000.dts |   23 +++++++++++++++++++----
 1 file changed, 19 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boot/dts/omap3-devkit8000.dts b/arch/arm/boot/dts/omap3-devkit8000.dts
index d6cff34..b4faddc 100644
--- a/arch/arm/boot/dts/omap3-devkit8000.dts
+++ b/arch/arm/boot/dts/omap3-devkit8000.dts
@@ -42,6 +42,19 @@
                 };
 
 	};
+
+	sound {
+		compatible = "ti,omap-twl4030";
+		ti,model = "devkit8000";
+
+		ti,mcbsp = <&mcbsp2>;
+		ti,codec = <&twl_audio>;
+		ti,audio-routing =
+			"Ext Spk", "PREDRIVEL",
+			"Ext Spk", "PREDRIVER",
+			"MAINMIC", "Main Mic",
+			"Main Mic", "Mic Bias 1";
+	};
 };
 
 &i2c1 {
@@ -50,6 +63,12 @@
 	twl: twl@48 {
 		reg = <0x48>;
 		interrupts = <7>;	/* SYS_NIRQ cascaded to intc */
+
+		twl_audio: audio {
+			compatible = "ti,twl4030-audio";
+			codec {
+			};
+		};
 	};
 };
 
@@ -85,10 +104,6 @@
 	status = "disabled";
 };
 
-&mcbsp2 {
-	status = "disabled";
-};
-
 &mcbsp3 {
 	status = "disabled";
 };
-- 
1.7.10.4

